lib/jsdom/living/xhr/XMLHttpRequestEventTarget-impl.js

```diff
@@ -1,10 +1,15 @@
 "use strict";
 const EventTargetImpl = require("../events/EventTarget-impl").implementation;
+const idlUtils = require("../generated/utils");
 const { setupForSimpleEventAccessors } = require("../helpers/create-event-accessor");
 
 const events = ["loadstart", "progress", "abort", "error", "load", "timeout", "loadend"];
 
-class XMLHttpRequestEventTargetImpl extends EventTargetImpl {}
+class XMLHttpRequestEventTargetImpl extends EventTargetImpl {
+  get _ownerDocument() {
+    return idlUtils.implForWrapper(this._globalObject._document);
```

Comment:
Can this ever change? If not (and I think not) then I'd prefer assigning it statically in the constructor for `XMLHttpRequestEventTargetImpl`.

More generally I feel this is a vestige of the days where we didn't have `._globalObject` available on every IDL object. But now that we do, maybe it's time for `EventTargetImpl` to use `this._globalObject._document` directly? Maybe something to think about later. @domenic @pmdartus

Reply:
The&nbsp;`_document`&nbsp;property gets&nbsp;cleared when&nbsp;`window.close()` is&nbsp;called.

This&nbsp;caused&nbsp;several WPT&nbsp;failures when&nbsp;I&nbsp;tried assigning&nbsp;it&nbsp;statically.

Reply:
I do agree there's probably a better way to do this these days, but we can do it in a separate PR. I'll push a TODO comment at least and then merge this.